public/app/core/utils/query.ts

```diff
@@ -9,6 +9,27 @@ export const getNextRefIdChar = (queries: DataQuery[]): string => {
   }
 };
 
+export function queryIsEmpty(query: DataQuery): boolean {
```

Comment:
This check does not feel future proof as it doesn't take new props into consideration. Any ideas of how to make this more robust?

Reply:
is this not the same as just doing Object.keys(query).length === 0 ?

Reply:
Currently this function is checking if the query has prop that is not defined in the `DataQuery` interface. So basically if it the data source has called `onChange` for the given query.